datafusion/core/src/physical_optimizer/enforce_sorting.rs

```diff
@@ -1053,6 +1074,28 @@ mod tests {
         Ok(())
     }
 
+    #[tokio::test]
+    async fn test_repartition_with_unecessary_preserve_order() -> Result<()> {
```

Comment:
I am not super happy with the test as it isn't a test of the optimizer, but the only time this can cause problems is when RepartitionExec::with_new_child is called, which I found really hard to
trigger with the optimizer tests.

Without the code in this PR, this test fails like:

```
[
    "SortPreservingRepartitionExec: partitioning=RoundRobinBatch(10), input_partitions=2",
    "  UnionExec",
    "    MemoryExec: partitions=1, partition_sizes=[0]",
    "    MemoryExec: partitions=1, partition_sizes=[0]",
]
actual:

[
    "RepartitionExec: partitioning=RoundRobinBatch(10), input_partitions=2",
    "  UnionExec",
    "    MemoryExec: partitions=1, partition_sizes=[0]",
    "    MemoryExec: partitions=1, partition_sizes=[0]",
]
```

Reply:
Is it possible to move this test to the same file with `struct RepartitionExec` as a unit test?